UI/src/app/app.component.ts

```diff
@@ -7,4 +10,16 @@ import { Component } from '@angular/core';
 })
 export class AppComponent {
   title = 'ANGULAR-GMP';
+
+  /**
+   *
+   */
+  constructor(private store: Store<RootState>) {
+
+  }
+  ngOnInit(): void {
+    //Called after the constructor, initializing input properties, and the first call to ngOnChanges.
+    //Add 'implements OnInit' to the class.
+    this.store.dispatch(getUserData());
```

Comment:
I just wondering is it works correctly?

As I see `AppComponent` doesn't implement `onInit` hook.

Also, I believe we don't need this hook and could dispatch an event from constructor because we don't have ane inputs here.

Reply:
https://angular.io/guide/lifecycle-hooks says that interfaces are optional. But i agree, that better to implement them.

> Interfaces are optional (technically)
> 
> The interfaces are optional for JavaScript and Typescript developers from a purely technical perspective. The JavaScript language doesn't have interfaces. Angular can't see TypeScript interfaces at runtime because they disappear from the transpiled JavaScript.
> 
> Fortunately, they aren't necessary. You don't have to add the lifecycle hook interfaces to directives and components to benefit from the hooks themselves.